store/copr/mpp.go

```diff
@@ -256,13 +278,21 @@ func (m *mppIterator) establishMPPConns(bo *tikv.Backoffer, req *kv.MPPDispatchR
 		return
 	}
 
-	// TODO: cancel the whole process when some error happens
 	for {
 		err := m.handleMPPStreamResponse(bo, resp, req)
 		if err != nil {
+			m.cancelMppTask(bo, taskMeta)
+			m.sendError(err)
+			return
+		}
+
+		if m.vars != nil && m.vars.Killed != nil && atomic.LoadUint32(m.vars.Killed) == 1 {
+			m.cancelMppTask(bo, taskMeta)
```

Comment:
cancelling the task during the loop is not enough. Considering if the stream hangs, this cancel will not work. A better way is to use 
the cancel function hooked in the stream. https://github.com/pingcap/tidb/blob/ae50555f8ab34a51170f5a40e4115ea3968bbe79/store/tikv/client.go#L489

Reply:
Considering if the stream hangs
-->
which stream?

Reply:
I mean grpc stream, we fetch the data packet continuely right? Given there is a dead loop in TiFlash, TiDB will hang on the Recv() function, right? That's the code that yet be covered by kill command.

Reply:
> Now that we have a mock mpp, why not write a unit test?

added.